compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -831,7 +831,14 @@ impl<'tcx> TypeFolder<TyCtxt<'tcx>> for ImplTraitInTraitCollector<'_, 'tcx> {
 
     fn fold_ty(&mut self, ty: Ty<'tcx>) -> Ty<'tcx> {
         if let ty::Alias(ty::Projection, proj) = ty.kind()
-            && self.interner().def_kind(proj.def_id) == DefKind::ImplTraitPlaceholder
+            && (self.interner().def_kind(proj.def_id) == DefKind::ImplTraitPlaceholder
+                || self
+                    .interner()
+                    .sess
+                    .opts
+                    .unstable_opts
+                    .lower_impl_trait_in_trait_to_assoc_ty
```

Comment:
I wonder if we should add a fn to `TyCtxt` that checks this flag, similar to [`TyCtxt::trait_solver_next`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.trait_solver_next)... `TyCtxt::lower_impl_trait_in_trait_to_assoc_ty`?

Reply:
Shorter than `.sess.opts.unstable_opts.blah.blahblah`